drivers/interrupt_controller/intc_plic.c

```diff
@@ -39,7 +40,17 @@
  * However, it is defined and supported by at least the Andes & Telink datasheet, and supported
  * in Linux's SiFive PLIC driver
  */
+#define PLIC_DRV_HAS_COMPAT(compat)                                                                \
+	DT_NODE_HAS_COMPAT(DT_COMPAT_GET_ANY_STATUS_OKAY(DT_DRV_COMPAT), compat)
+
+#if PLIC_DRV_HAS_COMPAT(andestech_plic)
+#define PLIC_REG_TRIG_TYPE_OFFSET 0x1080
```

Comment:
From what I remember, it isn't just the offset, but the number of bits per interrupt line. Some plic designs have zero bits of configuration (fully static, cannot even read configuration), some have 1 bit (level-only), and some have several bits (edge / level, rising / falling / both).

That should cover most of the base cases.